internal/common/connectionsapi/client_test.go

```diff
@@ -0,0 +1,292 @@
+package connectionsapi
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_sets_auth_token_and_content_type(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "Bearer some token", r.Header.Get("Authorization"))
+		assert.Equal(t, "application/json", r.Header.Get("Content-Type"))
+		_, _ = fmt.Fprintf(w, `{}`)
+	}))
+	defer svr.Close()
+
+	c, err := NewClient("some token", svr.URL, nil)
+	require.NoError(t, err)
+	_, err = c.CreateMetricsEndpointScrapeJob(context.Background(), "some stack id", MetricsEndpointScrapeJob{})
+	require.NoError(t, err)
+}
+
+func TestClient_CreateMetricsEndpointScrapeJob_sends_request_and_receives_response(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, http.MethodPost, r.Method)
+		assert.Equal(t, "/metrics-endpoint/stack/some-stack-id/jobs/test_job", r.URL.Path)
+		requestBody, err := io.ReadAll(r.Body)
+		require.NoError(t, err)
+		assert.JSONEq(t, `
+			{
+				"name":"test_job",
+				"enabled":true,
+				"authentication_method":"basic",
+				"basic_password":"my-password",
+				"basic_username":"my-username",
+				"url":"https://my-example-url.com:9000/metrics",
+				"scrape_interval_seconds":120
+			}`, string(requestBody))
+
+		w.WriteHeader(http.StatusCreated)
+		_, _ = w.Write([]byte(`
+			{
+				"status":"success",
+				"data":{
+					"test_job":{
+						"enabled":true,
+						"authentication_method":"basic",
+						"basic_username":"my-username",
+						"basic_password":"my-password",
+						"url":"https://my-example-url.com:9000/metrics",
+						"scrape_interval_seconds":120,
+						"flavor":"default"
+					}
+				}
+			}`))
```

Comment:
I believe this is what the current HE API is responding nowadays, are we keeping this structure?

Reply:
discussed this with Cristian and HE API should not be changed in particular